executor/index_lookup_join.go

```diff
@@ -556,6 +556,9 @@ func (iw *innerWorker) constructDatumLookupKey(task *lookUpJoinTask, chkIdx, row
 			if terror.ErrorEqual(err, types.ErrOverflow) {
 				return nil, nil
 			}
+			if terror.ErrorEqual(err, types.ErrTruncated) && (innerColType.Tp == mysql.TypeSet || innerColType.Tp == mysql.TypeEnum) {
```

Comment:
would `innerColType ` be `outerColType `?

Reply:
Only the inner side (index read side) will cause this problem.